lib/dio.dart

```diff
@@ -0,0 +1,94 @@
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:dio/dio.dart';
+import 'package:fullstory_flutter/fs.dart';
+
+/// Captures requests and responses from dio for display in Fullstory.
+class FSInterceptor extends Interceptor {
+  final _requestStartTimes = <Uri, int>{};
+
+  /// Computes the size (in bytes) of request data.
+  ///
+  /// Defaults to compute the size of the data's toString result in UTF8.
+  final int Function(RequestOptions) computeRequestSize;
+
+  /// Computes the size (in bytes) of response data.
+  ///
+  /// Defaults to compute the size of the data's toString result in UTF8.
+  final int Function(Response?) computeResponseSize;
+
+  /// Creates a [FSInterceptor] that captures network events.
+  ///
+  /// The [computeRequestSize] and [computeResponseSize] functions are used to
+  /// compute the size of the request and response data. By default, they
+  /// assime UTF-8 encoding. If your request uses another enocoding, override
+  /// these fields.
+  FSInterceptor({
+    this.computeRequestSize = requestFromUtf8,
+    this.computeResponseSize = responseFromUtf8,
+  });
+
+  @override
+  void onRequest(RequestOptions options, RequestInterceptorHandler handler) {
+    _requestStartTimes[options.uri] = DateTime.now().millisecondsSinceEpoch;
```

Comment:
Could there be multiple requests to the same URL in parallel? It looks like this could get confused, if so.

Reply:
Perhaps we should use an `Expando` with the request as the key :P

Reply:
Hmm, I like the expando idea. I looked through the code a bit, and there's a couple places where the options get deep copied though. I'll try it and see if the sequencing works for us.

Reply:
Works fine! Seems like the deep copies are early on and won't impact us. We'll have to keep a bit of an eye on that, but I don't think it's likely to change. There's a feature request to add round-trip time to the package, so hopefully that gets added and we can get rid of this at some point.

Reply:
Might be worth adding a comment about the deep copies and expando to give us a head start if it ever becomes a problem.